### PR TITLE
Move CES data store to @woocommerce/customer-effort-score

### LINKED ISSUE #37247: Move CES data store to @woocommerce/customer-effort-score
The CES data store that currently is found in the `woocommerce/admin/client` package should be moved to the `@woocommerce/customer-effort-score` package.

Location of data store files: `plugins/woocommerce/admin/client/customer-effort-score-tracks/data`

## Acceptance criteria

- [ ] Exports from `@woocommerce/customer-effort-score` are updated to allow `woocommerce/admin/client` to continue to build and use CES

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR moves the CES data store currently found in `woocommerce/admin/client` to the `@woocommerce/customer-effort-score` package. This builds on the work started in #37112.

**No functionality should be impacted by this change... this is just a package file refactoring.**

_Note: There are non-generic bits of functionality in the CES data store (controlling whether specific modals are shown, etc.)... these should be abstracted out in the future to a pattern where `@woocommerce/customer-effort-score` contains no feature-specific details. For now though, we can live with this._

Closes #37247.

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

Note: make sure tracking is enabled under WooCommerce > Settings > Advanced > WooCommerce.com

In order for CES snackbars and such to show up, you need to have not already dismissed/completed the CES for a particular feature. To reset things, you can delete the `woocommerce_ces_shown_for_actions` transient (use the WCA Test Helper).

1. Load this branch and run `pnpm install && pnpm run build` and make sure everything builds correctly.
2. Smoke test WCAdmin pages and make sure that any CES-related functionality works correctly.
   
    * Import CES snackbar - Go to Products > All Products; click "Import"; without continuing, click back to Products > All Products (see testing instructions on #36656)
    * Filtering snackbar - Use advanced filter in Analytics > Variations; enter advanced filter criteria and click "Filter"
    * CES footer on new product editor page - make sure "Share feedback" button brings up feedback modal.

3. Open your console and check if the following exist under `window.wc.customerEffortScore` (there will be more, these are the new ones added in this PR):

- `STORE_KEY`
- `QUEUE_OPTION_NAME`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/changelog/update-move-ces-data-store [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+dd CES data store to @woocommerce/customer-effort-score
```
Comment (nathanss) [inline plugins/woocommerce/changelog/update-move-ces-data-store]: Missing 'a'

**Review by nathanss — CHANGES_REQUESTED** [review]

**Review by nathanss — COMMENTED** [review]
While testing I found this odd behavior with the snackbar in the Analytics page, where I had to scroll all the way down to see it (it wasn't showing in a fixed position). I've seen and reported it before and I think we concluded that it was not our fault. Definitely not related to this PR.

File: plugins/woocommerce/changelog/update-move-ces-data-store (hunk 1 above)
Comment (mattsherman) [reply plugins/woocommerce/changelog/update-move-ces-data-store]: Oops! Fixed in 6e2ed19

**Comment by mattsherman** [comment]
@nathanss Fixed the typo -- ready for re-review!

> While testing I found this odd behavior with the snackbar in the Analytics page, where I had to scroll all the way down to see it (it wasn't showing in a fixed position). I've seen and reported it before and I think we concluded that it was not our fault. Definitely not related to this PR.

Hmm, haven't seen that! Do you know if we have an issue logged for it?

**Review by nathanss — APPROVED** [review]

**Comment by nathanss** [comment]
I think @louwie17 looked into it. Can you remember, Lourens?

**Pull request merged**